File: CS/Dashboard_AddCustomSettings/Form1.cs
using System;
using System.Windows.Forms;
using System.Drawing;
using System.IO;
using DevExpress.XtraBars;
using DevExpress.XtraBars.Ribbon;
using DevExpress.XtraCharts;
using DevExpress.DashboardWin;

namespace Dashboard_AddCustomSettings {
    public partial class Form1 : RibbonForm {
        bool dashboardChanged;
        ChartControl chartControl;

        public Form1() {
            InitializeComponent();
            dashboardDesigner1.DashboardItemControlCreated += 
                dashboardDesigner1_DashboardItemControlCreated;
            dashboardDesigner1.DashboardItemControlUpdated += 
                dashboardDesigner1_DashboardItemControlUpdated;
            dashboardDesigner1.DashboardItemBeforeControlDisposed += 
                dashboardDesigner1_DashboardItemBeforeControlDisposed;
            dashboardDesigner1.LoadDashboard(@"..\..\Data\Dashboard.xml");
            ribbonControl1.SelectedPage = chartDesignRibbonPage1;
        }

        void dashboardDesigner1_DashboardItemControlCreated(object sender, 
            DashboardItemControlEventArgs e) {
            if (e.DashboardItemName == "chartDashboardItem1") {
                chartControl = e.ChartControl;
            }
        }

        void dashboardDesigner1_DashboardItemControlUpdated(object sender, 
            DashboardItemControlEventArgs e) {
            if (e.DashboardItemName == "chartDashboardItem1") {
                ChartControl chartControl = e.ChartControl as ChartControl;
                barEditItem1.EditValue = chartControl.Legend.Font.Size;
            }
        }

        void dashboardDesigner1_DashboardItemBeforeControlDisposed(object sender, 
            DashboardItemControlEventArgs e) {
            if (e.DashboardItemName == "chartDashboardItem1") {
                chartControl = null;
            }
        }

        private void barEditItem1_EditValueChanged(object sender, EventArgs e) {
            if (chartControl != null) {
                float fontSize = Convert.ToSingle(((BarEditItem)sender).EditValue);
                chartControl.Legend.Font = new Font(chartControl.Legend.Font.FontFamily, fontSize);
                dashboardChanged = true;
            }
        }

        private void dashboardDesigner1_DashboardClosing(object sender, DashboardClosingEventArgs e) {
            if (dashboardChanged) {
                e.IsDashboardModified = true;
                e.Dashboard.UserData =
                    new System.Xml.Linq.XElement("chartFontSize", barEditItem1.EditValue);
                SaveDashboards();
            }
        }

        private void dashboardDesigner1_DashboardSaving(object sender, DashboardSavingEventArgs e) {
            if (e.Command == DashboardSaveCommand.Save && dashboardChanged) {
                e.Dashboard.UserData =
                    new System.Xml.Linq.XElement("chartFontSize", barEditItem1.EditValue);
                SaveDashboards();
            }
        }

        private void SaveDashboards() {
            string path = Directory.GetParent(Application.ExecutablePath).Parent.Parent.Parent.FullName;
            dashboardDesigner1.Dashboard.SaveToXml(path +
                "\\Dashboard_LoadCustomSettings_Win\\Data\\Dashboard.xml");
            dashboardDesigner1.Dashboard.SaveToXml(path +
                "\\Dashboard_LoadCustomSettings_Web\\App_Data\\Dashboard.xml");
            dashboardDesigner1.Dashboard.SaveToXml(path +
                "\\Dashboard_LoadCustomSettings_Mvc\\App_Data\\Dashboard.xml");
            dashboardChanged = false;
        }
    }
}

File: VB/Dashboard_AddCustomSettings/Form1.vb
Imports System
Imports System.Windows.Forms
Imports System.Drawing
Imports System.IO
Imports DevExpress.XtraBars
Imports DevExpress.XtraBars.Ribbon
Imports DevExpress.XtraCharts
Imports DevExpress.DashboardWin

Namespace Dashboard_AddCustomSettings
    Partial Public Class Form1
        Inherits RibbonForm

        Private dashboardChanged As Boolean
        Private chartControl As ChartControl

        Public Sub New()
            InitializeComponent()
            AddHandler dashboardDesigner1.DashboardItemControlCreated,
                AddressOf dashboardDesigner1_DashboardItemControlCreated
            AddHandler dashboardDesigner1.DashboardItemControlUpdated,
                AddressOf dashboardDesigner1_DashboardItemControlUpdated
            AddHandler dashboardDesigner1.DashboardItemBeforeControlDisposed,
                AddressOf dashboardDesigner1_DashboardItemBeforeControlDisposed
            dashboardDesigner1.LoadDashboard("..\..\Data\Dashboard.xml")
            ribbonControl1.SelectedPage = chartDesignRibbonPage1
        End Sub

        Private Sub dashboardDesigner1_DashboardItemControlCreated(ByVal sender As Object, _
                                       ByVal e As DashboardItemControlEventArgs)
            If e.DashboardItemName = "chartDashboardItem1" Then
                chartControl = e.ChartControl
            End If
        End Sub

        Private Sub dashboardDesigner1_DashboardItemControlUpdated(ByVal sender As Object, _
                                       ByVal e As DashboardItemControlEventArgs)
            If e.DashboardItemName = "chartDashboardItem1" Then
                Dim chartControl As ChartControl = TryCast(e.ChartControl, ChartControl)
                barEditItem1.EditValue = chartControl.Legend.Font.Size
            End If
        End Sub

        Private Sub dashboardDesigner1_DashboardItemBeforeControlDisposed(ByVal sender As Object, _
                                       ByVal e As DashboardItemControlEventArgs)
            If e.DashboardItemName = "chartDashboardItem1" Then
                chartControl = Nothing
            End If
        End Sub

        Private Sub barEditItem1_EditValueChanged(ByVal sender As Object, _
                                 ByVal e As EventArgs) Handles barEditItem1.EditValueChanged
            If chartControl IsNot Nothing Then
                Dim fontSize As Single = Convert.ToSingle(DirectCast(sender, BarEditItem).EditValue)
                chartControl.Legend.Font = New Font(chartControl.Legend.Font.FontFamily, fontSize)
                dashboardChanged = True
            End If
        End Sub

        Private Sub dashboardDesigner1_DashboardClosing(ByVal sender As Object, _
                                        ByVal e As DashboardClosingEventArgs) _
                                    Handles dashboardDesigner1.DashboardClosing
            If dashboardChanged Then
                e.IsDashboardModified = True
                e.Dashboard.UserData =
                    New System.Xml.Linq.XElement("chartFontSize", barEditItem1.EditValue)
                SaveDashboards()
            End If
        End Sub

        Private Sub dashboardDesigner1_DashboardSaving(ByVal sender As Object, _
                                       ByVal e As DashboardSavingEventArgs) _
                                   Handles dashboardDesigner1.DashboardSaving
            If e.Command = DashboardSaveCommand.Save AndAlso dashboardChanged Then
                e.Dashboard.UserData =
                    New System.Xml.Linq.XElement("chartFontSize", barEditItem1.EditValue)
                SaveDashboards()
            End If
        End Sub

        Private Sub SaveDashboards()
            Dim path As String =
                Directory.GetParent(Application.ExecutablePath).Parent.Parent.Parent.FullName
            dashboardDesigner1.Dashboard.SaveToXml(path & "\Dashboard_LoadCustomSettings_Win\Data\Dashboard.xml")
            dashboardDesigner1.Dashboard.SaveToXml(path & "\Dashboard_LoadCustomSettings_Web\App_Data\Dashboard.xml")
            dashboardDesigner1.Dashboard.SaveToXml(path & "\Dashboard_LoadCustomSettings_Mvc\App_Data\Dashboard.xml")
            dashboardChanged = False
        End Sub
    End Class
End Namespace